src/shared/components/ProfilePage/Stats/DistributionGraph/index.jsx

```diff
@@ -0,0 +1,288 @@
+/* eslint-env browser */
+import d3 from 'd3';
+import React from 'react';
+import { toPairs } from 'lodash';
+import PT from 'prop-types';
+import ChartTooltip from '../ChartTooltip';
+import './index.scss';
+
+const getRanges = distribution => (
+  toPairs(distribution)
+    .map(([name, number]) => ({
+      name,
+      number,
+      start: parseInt(name.match(/\d+/)[0], 10),
+      end: parseInt(name.match(/To(\d+)/)[1], 10),
+    }))
+    .sort((a, b) => a.start - b.start)
+);
+
+function removeLeadingZeroes(ranges) {
+  while (ranges[0].number === 0) {
+    ranges.shift();
+  }
```

Comment:
Bad. Iterative popping from beginning of an array is inefficient. If such operation is necessary, one should find the first non-zero index `i`, then do a single `slice(i)` on the array.